src/main/java/com/marginallyclever/makelangelo/plotter/plotterControls/PlotterControls.java

```diff
@@ -90,12 +90,16 @@ private void onMarlinEvent(ActionEvent e) {
 		switch (e.getActionCommand()) {
 		case MarlinInterface.IDLE ->
 				{ if (isRunning) step(); }
-		case MarlinInterface.ERROR ->
-				JOptionPane.showMessageDialog(this, Translator.get("PlotterControls.Error"), Translator.get("ErrorTitle"),  JOptionPane.ERROR_MESSAGE);
+		case MarlinInterface.ERROR,
+			 MarlinInterface.DID_NOT_FIND,
+			 MarlinInterface.COMMUNICATION_FAILURE -> {
+			if (!isErrorAlreadyDisplayed) {
+				JOptionPane.showMessageDialog(this, Translator.get("PlotterControls." + e.getActionCommand()), Translator.get("ErrorTitle"), JOptionPane.ERROR_MESSAGE);
+				isErrorAlreadyDisplayed = true;
+			}
+		}
```

Comment:
test here that MarlinPlotterInterface remembers the line number (A) so that after reconnect and home it can resume from line (A-1).  but how to automate such a test?

Reply:
hum quite challenging, I'm opening an issue with this idea 💪